src/linking/linking.cpp

```diff
@@ -1460,7 +1467,7 @@ void linkingt::copy_symbols(
   }
 
   // Move over all the non-colliding ones
-  std::unordered_set<irep_idt> collisions;
+  std::unordered_set<irep_idt> type_collisions, non_type_collisions;
```

Comment:
It's not clear to me there is significant value in having different collections.

Reply:
The entries in the sets require different methods to be called (`duplicate_type_symbol` vs `duplicate_non_type_symbol`), and the order of invocation matters (see commit message). Hence the two collections.